tests/unit/semaphore_test.cc

```diff
@@ -452,3 +452,58 @@ SEASTAR_THREAD_TEST_CASE(test_reassigned_units_are_returned) {
     wait.get();
     t.cancel();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_get_units_after_move) {
```

Comment:
just curious - did *all* these tests crash before the fix?

Reply:
yes, they all fail due to the same root cause from slightly different paths.